### PR TITLE
Document how to use elementRef

### PR DESCRIPTION
**Pull request by jez**
### Summary & motivation

While we had documented the type of `elementRef`, we didn't have a good example
of how to use it or what kinds of problems it can be used to solve.

This PR adds a simple example of how to use `elementRef` to access the
underlying Element from Stripe.js to focus the input on page load.

### CODE REVIEW

**Comment by jez** [comment]
cc @fred-stripe 

r? @atty-stripe

File: README.md [hunk 1]
```diff
@@ -537,6 +537,38 @@ type PaymentRequestButtonProps = {
 };
 ```
 
+#### Using `elementRef`
+
+Note that the `elementRef` is a sort of escape hatch in case you need access to
+the underlying Element object from Stripe.js. For example, you can use
+`elementRef` to force your element to focus:
```
Comment (fred-stripe) [inline README.md]: Do you think it's worth pointing out that this is similar to the `refs` pattern in React itself?[1] Two details that I think need to be added here:

1/ Clarify that the first argument passed `elementRef` is the underlying Element.
2/ Be explicit about when the `elementRef` function is called. It looks like it's called when the Element is 
 mounted and ready.[2]

Given [2] we could simplify the example below by removing `handleReady` and using `elementRef={(el) => el.focus()}` but that locks us into the current lifecycle behavior unless we make an explicit guarantee about the `ready`-ness of the passed element.

If we guarantee that the passed Element will always be `ready` that gives us a strong contract we can be explicit about, which makes it easier to reason about this property. This _seems_ like a reasonable guarantee but I'm having trouble imagining a possible future or changes where we wouldn't want that to be the case—thinking aloud, can you do anything useful to an Element _before_ its ready? I suppose you can call `el.update()`[3] but you'd be better off doing so with props on the React component.

Tl;dr: if we think that we can guarantee that the Element will always be `ready` when `elementRef` is called, then we should explicitly say so & update the example accordingly.

[1] https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-dom-element
[2] https://github.com/stripe/react-stripe-elements/blob/bf6a45fc036b7ec8821149d3def997a181ae8bf8/src/components/Element.js#L102
[3] https://stripe.com/docs/stripe-js/reference#other-methods

File: README.md [hunk 2]
```diff
@@ -537,6 +537,38 @@ type PaymentRequestButtonProps = {
...
+  handleReady = () => {
+    this._element.focus();
+  };
+
+  render = () => {
+    return (
+      <label>
+        Card details
+        <CardElement
+          elementRef={(el) => {this._element = el}}
```
[the remark this answers is not in the document]
Comment (jenan-stripe) [reply README.md]: Strings refs are semi-deprecated: https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs

File: README.md [hunk 3]
```diff
@@ -537,6 +537,38 @@ type PaymentRequestButtonProps = {
...
+          onReady={this.handleReady}
```
[the remark this answers is not in the document]
Comment (jez) [reply README.md]: What do you mean? It's defined in this example 10 lines up. It's also necessary, so that we can call `.focus()` when the element has been mounted in the page.

File: README.md (hunk 1 above)
Comment (jez) [reply README.md]: I didn't think about this hard enough when I first sat down with it.
I think I'll revisit it and see what we should do.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (atty-stripe) [reply README.md]: Ah sorry my mistake.

File: README.md (hunk 1 above)
Comment (jez) [reply README.md]: This is really strange. This is where `elementRef` is called:

```js
element.on('ready', () => {
  this.props.elementRef(this._element);
  this.props.onReady();
});
```

So we could have left out `elementRef` and just modified the signature of `onReady` to provide the underlying Element.

I think I'm going to open an API review for this and see what people think.
